arches_controlled_lists/models.py

```diff
@@ -185,14 +187,28 @@ class ListItem(models.Model):
     class Meta:
         ordering = ["sortorder"]
         constraints = [
-            # Sort order concerns the list as a whole, not subsets
```

Comment:
notes on downstream impact (I can grab these if you want):
- will need to check knockout reference select widget, ensure flat items list is not mangled now.
- will need to update `.annotate(max_sortorder=Max("list_items__sortorder", default=-1))`
- will need to update `lookup_listitem_from_label` to make it deterministic again
- consider simplifying the vue code that handles reordering operations